### ISSUE TITLE
(feature idea) Compute terminal bounds per tab rather than per session

### ISSUE THREAD

**Issue by mathieulj** [zellij-org/zellij#3056]
I regularly use Zellij to maintain a session across multiple open terminals and would love to have tabs that can compute their sizes taking into consideration which clients are viewing them. The most typical reason to have multiple terminals connected is to have a drop down terminal (ex. [guake](https://github.com/Guake/guake)/quake) to invoke commands or tailing logs while interacting with the gui (ex. the app I'm developing) and a fullscreen terminal on a dedicated desktop for focused work. I often end up starting two zellij sessions so that both can be optimized for their respective sizes but that can be annoying as I have to toggle between sessions (ex. if I want to see logs, I'd have to connect to whichever session started the app).

I haven't yet made myself familiar with the Zellij code base but I work with Rust daily and I'm willing to explore prototyping this idea if the change in behaviour would be welcome. Hints about code structure and/or where to look to find the existing per session logic would be appreciated if that were the case.

TLDR; I'd love for two clients connected to the same session but viewing different tabs to each have their terminal sizes respected even if they don't match. The existing behaviour (minimum dimensions) would be retained when multiple clients are viewing the same tab.

**Comment by mathieulj** [comment]
Swinging back to this one. I assumed the self assignment was an expressed intent to work on the issue yourself @imsnif. Was that assumption incorrect? I have not looked into this since I put up the issue because of that assumption.

**Comment by imsnif** [comment]
Yes, but I unfortunately cannot provide an ETA.
